### PR TITLE
Add option `packet-data-query-height` to Hermes `packet-recv` CLI

### PR DESCRIPTION
**Pull request by adizere**
Closes: #2453

## Description

The fix in this dev branch was already [validated in practice](https://github.com/informalsystems/ibc-rs/issues/2453#issuecomment-1200396456), having proved to work with timing-out four packets that were stuck for +1 month on the Evmos <> Osmosis channel (see #2453 for more details).

We just need to clean-up the branch.

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: relayer/src/chain/cosmos/query/tx.rs [hunk 1]
```diff
@@ -36,11 +37,40 @@ pub async fn query_txs(
...
+                    result.append(
+                        &mut tx_copy
+                            .events
+                            .into_iter()
+                            .filter_map(|e| filter_matching_event(e, &request, &request.sequences))
+                            .collect(),
+                    )
+                }
+            }
+            warn!(
```
Comment (adizere) [inline relayer/src/chain/cosmos/query/tx.rs:69]: Likely this is the only cleanup we need to do. Plus the changelog.

File: relayer-cli/src/commands/tx/packet.rs [hunk 2]
```diff
@@ -48,6 +50,14 @@ pub struct TxPacketRecvCmd {
         help = "Identifier of the source channel"
     )]
     src_channel_id: ChannelId,
+
+    #[clap(
+        long = "packet-data-query-height",
+        required = false,
+        help_heading = "NOT REQUIRED",
```
Comment (adizere) [inline relayer-cli/src/commands/tx/packet.rs:57]: We should maybe make this prettier, unclear what is does exactly and if its useful in more places.

File: relayer-cli/src/commands/tx/packet.rs [hunk 3]
```diff
@@ -48,6 +50,14 @@ pub struct TxPacketRecvCmd {
         help = "Identifier of the source channel"
     )]
     src_channel_id: ChannelId,
+
+    #[clap(
+        long = "packet-data-query-height",
+        required = false,
```
Comment (adizere) [inline relayer-cli/src/commands/tx/packet.rs:56]: Luca: This is likely not required.

**Comment by adizere** [comment]
Discussed with Luca and he can take over this once #2505 and #2537 are done, and once v1 is out.

File: relayer-cli/src/commands/tx/packet.rs (hunk 2 above)
Comment (ljoss17) [reply relayer-cli/src/commands/tx/packet.rs:57]: The `help_heading` is used to specify a category for the flag, so with this one set to `NOT REQUIRED` the help output is:
```
hermes-tx-packet-recv 
Relay receive or timeout packets

USAGE:
    hermes tx packet-recv [OPTIONS] --dst-chain <DST_CHAIN_ID> --src-chain <SRC_CHAIN_ID> --src-port <SRC_PORT_ID> --src-channel <SRC_CHANNEL_ID>

OPTIONS:
    -h, --help    Print help information

REQUIRED:
        --dst-chain <DST_CHAIN_ID>        Identifier of the destination chain
        --src-chain <SRC_CHAIN_ID>        Identifier of the source chain
        --src-channel <SRC_CHANNEL_ID>    Identifier of the source channel [aliases: src-chan]
        --src-port <SRC_PORT_ID>          Identifier of the source port

NOT REQUIRED:
        --packet-data-query-height <PACKET_DATA_QUERY_HEIGHT>
            Height at which the packet data is queried
```

But the CLIs all have the optional flags under the `OPTIONS` category, so I will remove the `help_heading` in order to have the help output:
```
hermes-tx-packet-recv 
Relay receive or timeout packets

USAGE:
    hermes tx packet-recv [OPTIONS] --dst-chain <DST_CHAIN_ID> --src-chain <SRC_CHAIN_ID> --src-port <SRC_PORT_ID> --src-channel <SRC_CHANNEL_ID>

OPTIONS:
    -h, --help
            Print help information

        --packet-data-query-height <PACKET_DATA_QUERY_HEIGHT>
            Height at which the packet data is queried

REQUIRED:
        --dst-chain <DST_CHAIN_ID>        Identifier of the destination chain
        --src-chain <SRC_CHAIN_ID>        Identifier of the source chain
        --src-channel <SRC_CHANNEL_ID>    Identifier of the source channel [aliases: src-chan]
        --src-port <SRC_PORT_ID>          Identifier of the source port
```

File: relayer-cli/src/commands/tx/packet.rs (hunk 3 above)
Comment (ljoss17) [reply relayer-cli/src/commands/tx/packet.rs]: Confirmed it is not necessary

**Review by seanchen1991 — APPROVED** [review]
Looks good to me 🙂

**Pull request merged**